src/puffin/src/file_metadata.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use derive_builder::Builder;
+use serde::{Deserialize, Serialize};
+
+use crate::blob_metadata::BlobMetadata;
+
+/// Metadata of a Puffin file
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Builder)]
+pub struct FileMetadata {
+    /// Metadata for each blob in the file
+    #[builder(default)]
+    #[serde(rename = "blobs")]
+    pub blob_metadata: Vec<BlobMetadata>,
```

Comment:
question: Why not name it as `blobs`?

Reply:
Small taste issue, I feel like `blobs` is misleading as actual blobs rather than metadata, but it doesn’t matter, I’ve changed it